### PR TITLE
Talk about when to double number of bins

### LINKED ISSUE #14: Number of bins
How many do we need for # of targets a year from now?

### PR DESCRIPTION
**Pull request by trishankatdatadog**
Fix #14

### CODE REVIEW

File: pep-0458.txt [hunk 1]
```diff
@@ -457,6 +457,17 @@ pip for the first time.
 
 __ https://docs.google.com/spreadsheets/d/11_XkeHrf4GdhMYVqpYWsug6JNz5ZK6HvvmDZX0__K2I/edit?usp=sharing
 
+This number of bins SHOULD double when the metadata overhead for returning
+users exceeds 50% (relative to the average size of downloaded packages).
+Presently, this SHOULD happen when the number of targets (simple indices
+and packages) increase at least 4x from over 2M to nearly 9M, at which
+point the metadata overhead for returning and new users would be around
+49-54% and 185% respectively, assuming that the number of bins stay fixed.
```
Comment (mnm678) [inline pep-0458.txt]: Are these the worst case numbers? I think we should make that clear because it might affect when doubling should occur in practice

Comment (trishankatdatadog) [reply pep-0458.txt]: Sorry, could you clarify what "worst case numbers" means?

Comment (mnm678) [reply pep-0458.txt]: They assume that all filenames, etc are the maximum length

File: pep-0458.txt [hunk 2]
```diff
@@ -457,6 +457,17 @@ pip for the first time.
 
 __ https://docs.google.com/spreadsheets/d/11_XkeHrf4GdhMYVqpYWsug6JNz5ZK6HvvmDZX0__K2I/edit?usp=sharing
 
+This number of bins SHOULD double when the metadata overhead for returning
+users exceeds 50% (relative to the average size of downloaded packages).
+Presently, this SHOULD happen when the number of targets (simple indices
+and packages) increase at least 4x from over 2M to nearly 9M, at which
```
Comment (lukpueh) [inline pep-0458.txt]: Are there other targets than simple indices or packages? I think it's not necessary to mention it, and this sentence might benefit from being a tad shorter. :)

File: pep-0458.txt [hunk 3]
```diff
@@ -457,6 +457,17 @@ pip for the first time.
 
 __ https://docs.google.com/spreadsheets/d/11_XkeHrf4GdhMYVqpYWsug6JNz5ZK6HvvmDZX0__K2I/edit?usp=sharing
 
+This number of bins SHOULD double when the metadata overhead for returning
+users exceeds 50% (relative to the average size of downloaded packages).
```
Comment (lukpueh) [inline pep-0458.txt]: Can you move up this note, i.e. "(relative to the average size of downloaded packages)", to when you first mention percentage of overhead in [L454](https://github.com/secure-systems-lab/peps/pull/29/files#diff-7ffbd592926c74789442ddd1465bb9e0R454)?

File: pep-0458.txt (hunk 1 above)
Comment (lukpueh) [reply pep-0458.txt]: I agree with Marina. But IMO pointing to the numbers, as done in [L458](https://github.com/secure-systems-lab/peps/pull/29/files#diff-7ffbd592926c74789442ddd1465bb9e0R458), should be enough.

File: pep-0458.txt [hunk 4]
```diff
@@ -457,6 +457,17 @@ pip for the first time.
...
+Presently, this SHOULD happen when the number of targets (simple indices
+and packages) increase at least 4x from over 2M to nearly 9M, at which
+point the metadata overhead for returning and new users would be around
+49-54% and 185% respectively, assuming that the number of bins stay fixed.
+If the number of bins is increased, then the cost for all users would
+effectively be the cost for new users. If the cost for new users should
```
Comment (lukpueh) [inline pep-0458.txt]: I don't understand why that's the case.  Could you explain?

**Review by lukpueh — COMMENTED** [review]
Looks good overall. I only have few minor comments.

File: pep-0458.txt (hunk 1 above)
Comment (trishankatdatadog) [reply pep-0458.txt]: OK, I'll clarify

File: pep-0458.txt (hunk 2 above)
Comment (trishankatdatadog) [reply pep-0458.txt]: OK

File: pep-0458.txt (hunk 3 above)
Comment (trishankatdatadog) [reply pep-0458.txt]: NP

File: pep-0458.txt (hunk 4 above)
Comment (trishankatdatadog) [reply pep-0458.txt]: Because the cost for new users is dominated by the `bins` metadata. Will make a note.

**Comment by trishankatdatadog** [comment]
@mnm678 @lukpueh can i plz haz your ✅? kthxbai

**Review by mnm678 — APPROVED** [review]
LGTM

**Pull request merged**